cmd/zed/api/credentials.go

```diff
@@ -0,0 +1,99 @@
+package api
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/brimdata/zed/pkg/fs"
+)
+
+const (
+	configDirName = "zapi"
+	credsFileName = "credentials.json"
+)
+
+func (c *Command) zapiPath() (string, error) {
+	if c.configDir != "" {
+		if err := os.MkdirAll(c.configDir, 0777); err != nil {
+			return "", err
+		}
+		return filepath.Abs(c.configDir)
+	}
+	dir, err := os.UserConfigDir()
+	if err != nil {
+		return "", err
+	}
+	zcd := filepath.Join(dir, configDirName)
+	if err := os.MkdirAll(zcd, 0777); err != nil {
+		return "", err
+	}
+	return zcd, nil
+}
+
+func (c *Command) LoadCredentials() (*Credentials, error) {
+	path := filepath.Join(c.configDir, credsFileName)
+	var creds Credentials
+	if err := fs.UnmarshalJSONFile(path, &creds); err != nil {
+		if os.IsNotExist(err) {
+			return &Credentials{}, nil
+		}
+		return nil, err
+	}
+	return &creds, nil
+}
+
+func (c *Command) SaveCredentials(creds *Credentials) error {
+	if err := os.MkdirAll(c.configDir, 0777); err != nil {
```

Comment:
This stuff is private!
```suggestion
	if err := os.MkdirAll(c.configDir, 0700); err != nil {
```